webauthn-server-core/src/main/java/com/yubico/webauthn/OriginMatcher.java

```diff
@@ -0,0 +1,51 @@
+package com.yubico.webauthn;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Set;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+class OriginMatcher {
+
+    static boolean isAllowed(
+        String origin,
+        Set<String> allowedOrigins,
+        boolean allowPort,
+        boolean allowSubdomain
+    ) {
+        return allowedOrigins.stream().anyMatch(allowedOriginString -> {
+            if (allowedOriginString.equals(origin)) {
+                return true;
+            } else {
+                final URL allowedOrigin;
+                try {
+                    allowedOrigin = new URL(allowedOriginString);
+                } catch (MalformedURLException e) {
```

Comment:
It would be helpful to print out the stacktrace or at least a log message describing the error

Reply:
Yeah, good point, thanks!. I had some thought of parsing the `allowedOriginString`s earlier on and fail at startup instead of when processing a request. I'll take another look at if I can do that without breaking the API.

Reply:
Knowing which string that caused the error could be even more helpful. But actually, any indication of where the error originated is a huge help for someone debugging an error.